File: Problems/0100_0199/0106_Construct_Binary_Tree_from_Inorder_and_Postorder_Traversal/Project_Java/Mylib.java
public class Mylib {
    public int[] stringTointArray(String s) {
        if (s.length() <= 0)
            return null;

        String[] flds = s.split(",");
        int[] nums = new int[flds.length];

        if (flds.length <= 0)
            return nums;

        for (int i = 0; i < nums.length; ++i) {
            nums[i] = Integer.parseInt(flds[i]);
        }

        return nums;
    }

    public String intArrayToString(int[] nums) {
        if (nums == null)
            return "";
        if (nums.length <= 0)
            return "";

        StringBuilder resultStr = new StringBuilder("[" +  Integer.toString(nums[0]));

        for (int i = 1; i < nums.length; ++i)
            resultStr.append("," + Integer.toString(nums[i]));

        return resultStr.append("]").toString();
    }
}

File: Problems/0100_0199/0144_Binary_Tree_Preorder_Traversal/Project_Go/solution/mylib.go
package solution

import (
	"strconv"
	"strings"
)

func IntArrayToString(nums []int) string {
	if len(nums) <= 0 {
		return ""
	}

	resultStr := strconv.Itoa(nums[0])
	for i := 1; i < len(nums); i++ {
		resultStr += ", " + strconv.Itoa(nums[i])
	}

	return resultStr
}

func IntIntArrayToString(nums [][]int) string {
	if len(nums) <= 0 {
		return "[]"
	}

	resultStr := "[[" + IntArrayToString(nums[0]) + "]"

	for i := 1; i < len(nums); i++ {
		resultStr += ", [" + IntArrayToString(nums[i]) + "]"
	}
	return resultStr + "]"
}

func StringToIntArray(flds string) []int {
	numsStr := strings.Split(flds, ",")
	nums := make([]int, len(numsStr))

	for i := 0; i < len(nums); i++ {
		nums[i], _ = strconv.Atoi(numsStr[i])
	}

	return nums
}
